Logstash/logstash-input-azurewadtable/lib/logstash/inputs/azurewadtable.rb

```diff
@@ -52,14 +53,34 @@ def run(output_queue)
   def teardown
   end  
 
-  def process(output_queue)
-    @logger.debug(@last_timestamp)
+  def build_latent_query
+    @logger.debug(@until_timestamp)
+    query_filter = "(PartitionKey gt '#{partitionkey_from_datetime(@last_timestamp)}' and PartitionKey lt '#{partitionkey_from_datetime(@until_timestamp)}')"
+    for i in 0..99
+      query_filter << " or (PartitionKey gt '#{i.to_s.rjust(19, '0')}___#{partitionkey_from_datetime(@last_timestamp)}' and PartitionKey lt '#{i.to_s.rjust(19, '0')}___#{partitionkey_from_datetime(@until_timestamp)}')"
+    end # for block
+    query_filter = query_filter.gsub('"','')
+    query_filter
+  end
+
+  def build_zero_latency_query
```

Comment:
This scenario is to support single-VM data sets where users should not be subjected to a delay in retrieving their data.